### PR TITLE
Add notice to table when there is no profile data

### LINKED ISSUE #190: Redirect to Hours/Submission
The profile will probably be left unedited once set. We should either redirect people to their hours or submission once logged into the website.

### PR DESCRIPTION
**Pull request by theTisch21**
This PR will:
- Resolve eastlakehs/community-service-tracker#190 
- Change it to redirect to the hours table instead of the profile page after login
- Make it so that if the user does not have their profile data, redirect them to the profile page after login

I'm aware formatting is wrong, I'll run prettier before marking this ready for review

### CODE REVIEW

**Comment by Zjjc123** [comment]
@theTisch21 The issue is that the data isn't updated instantly, so it will always be "" until it is updated.

**Comment by theTisch21** [comment]
I haven't tested these latest commits. I'll do that later today

**Comment by theTisch21** [comment]
It's tested and ready!
# This PR:
- Changes the login system to redirect to the table instead of the profile page
- Change the "/" route to redirect to the table instead of the profile page
- If a user has not set their profile data, they will be presented a message on the table directing them to update their profile data
- Closes #190

File: website/src/components/table/table.tsx [hunk 1]
```diff
@@ -80,6 +82,16 @@ const CSTable: React.FunctionComponent<{
     });
   };
 
+  if (!userData.firstName) {
```
Comment (daniel-sudz) [inline website/src/components/table/table.tsx:85]: I can't remember how we implemented this but is it possible for the first name to be set but the graduation year to not be set?

**Review by daniel-sudz — APPROVED** [review]
One thing that might be annoying is that the profile data is on a listener and might load after the page is visited. This could make an annoying flash of content on missing profile data before showing the table. We could solve this by differentiating between a missing profile and a profile that has not gotten data yet (default to table view).

**Comment by daniel-sudz** [comment]
btw @theTisch21 thanks for the changes especially given our messy code base lol

**Comment by theTisch21** [comment]
@daniel-sudz 
Is it possible to set it so that there's a quick flash of the table before the profile data message? Since they would only see that once, that would be better than a flash of no profile before content. 
I tried to add another field to accounts to mark them as having profile data or not, but I couldn't figure out how to do that.

**Comment by Zjjc123** [comment]
How did you test this? The only way I'm doing currently is going into firebase console and just manually removing my profile

File: website/src/components/table/table.tsx (hunk 1 above)
Comment (Zjjc123) [reply website/src/components/table/table.tsx:85]: yes I have tested that and that is the case. you don't need a grad year

**Comment by Zjjc123** [comment]
Could we also do the same thing for the submit page?

**Comment by theTisch21** [comment]
> Could we also do the same thing for the submit page?

Sure, I'll copy the code once I implement Daniel's request.

- [ ] Copy the code to the submit page

**Comment by theTisch21** [comment]
> How did you test this? The only way I'm doing currently is going into firebase console and just manually removing my profile
> 

@Zjjc123 
I created a fire base app myself and enabled the database. My testing is WAY below the free limits, so I haven't had to pay for anything. I just go to the console and delete my testing profile to test it, then I sign out and back in. Once we get authentication stubbed in development (#175), I'll just use that instead.

File: website/src/components/table/table.tsx (hunk 1 above)
Comment (theTisch21) [reply website/src/components/table/table.tsx:85]: Should I have it check for name and grad year? Or is just checking for name enough?

Comment (Zjjc123) [reply website/src/components/table/table.tsx:85]: grad year would be great

**Comment by theTisch21** [comment]
I thought I'd take another crack at this, but I can't even log in anymore, even on development. I'm closing this for now, if I ever figure out how to log in again I'll finish this up.

**Pull request closed without merging**